src/wpool/mongoose_gun_worker.erl

```diff
@@ -0,0 +1,262 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% This module a wpool worker capable of establishing HTTP/2
+%%% connections using the `Gun' library.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_gun_worker).
+
+-behaviour(gen_server).
+
+%% API
+-export([start_link/2]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_continue/2,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2]).
```

Comment:
We should implement some termination, to clean-up any alive connection, in the line of

```
terminate(_Reason, #state{requests = Requests}) ->
     maps:fold(
         fun(StreamRef, {_Request, #response_data{from = From, timeout_timer = TRef}}, _) ->
               gun:flush(StreamRef),
               erlang:cancel_timer(TRef),
               gen_server:reply(From, {error, some_error}),
               ok
        end, ok, Requests).
```